src/cli_plugin/install/project_dependencies.js

```diff
@@ -0,0 +1,41 @@
+import path from 'path';
+
+import * as fs from '../lib/fs';
+
+const LINK_DEP = 'link:';
+
+const isKibanaDep = depVersion =>
+  depVersion.startsWith(`${LINK_DEP}../../kibana/`);
+
+export async function prepareProjectDependencies(settings, logger) {
+  await handleLinkDependencies(
+    settings.workingPath,
+    logger
+  );
+}
+
+async function handleLinkDependencies(pluginPath) {
+  const packageJsonPath = path.resolve(pluginPath, 'package.json');
+  const rawPkgJson = await fs.readFile(packageJsonPath, 'utf-8');
+  const pkgJson = JSON.parse(rawPkgJson);
+
+  const deps = pkgJson.dependencies || {};
+
+  for (const depName of Object.keys(deps)) {
+    const depVersion = deps[depName];
+
+    if (depVersion.startsWith(LINK_DEP)) {
+      if (isKibanaDep) {
```

Comment:
Haha 🙈 I've added some tests for it. And fixed the `if` too (initially did it like that because I had code in the "kibana-if" too, but yeah, this is cleaner)